### ISSUE TITLE
Add more decimate options

### ISSUE THREAD

**Issue by jbednar** [ioam/holoviews#2934]
Right now, the HoloViews `decimate` operator randomly samples from the available data, dynamically on each zoom or pan.  This is often a useful behavior, but there are other options that can be useful for working with large datasets:

- [ ] **Regular sampling**: Random sampling is not very useful for Elements like Curve and Path, where regular sampling would be a better default for keeping the overall shape
- [ ] **Outlier preserving**: Some algorithms like [Largest Triangle](https://blog.scottlogic.com/2015/11/16/sampling-large-data-in-d3fc.html) are better at preserving outliers
- [ ] Others?

If we implemented multiple types of sampling, the policy could be selected by `mode='random'`, `mode='regular'`, etc.  We might want the default to be `mode=None`, where a mode appropriate to that Element type (e.g. regular for Curve) would be selected.

**Comment by ayoungs** [comment]
I'm definitely a fan of the slicing with stride downsampling option!

**Comment by jlstevens** [comment]
Not sure about the other two suggestions but regular downsampling should be a quick, easy, understandable and useful way to decimate instead of always applying decimation randomly.

**Comment by RiccardoGiro** [comment]
Did you manage to implement any of those features?

**Comment by jbednar** [comment]
No, it's wide open for any interested user to dive into, and should be an easy win!

**Comment by jvdd** [comment]
Hey @jbednar

I wanted to let you know about a new library that I have developed called [tsdownsample](https://github.com/predict-idlab/tsdownsample). It is an optimized implementation of various time series downsampling algorithms written in Rust, and it is available as a stand-alone Python package. The goal of this library was to improve the efficiency of these algorithms, while also allowing them to be used independently of plotly-resampler.

The library is still in the early stages of development, but it is beginning to mature. I am reaching out to see if there is interest in using tsdownsample to provide time series decimation options. If so, I would be happy to submit a pull request. Additionally, I welcome any feedback you may have on the interface for the downsample methods. More details can be found below :arrow_down: 

The signature of the downsample methdods:
```
downsample([x], y, n_out, **kwargs)
```
Note the following:
- `x` is optional
- `x` and `y` are both positional arguments (this is in line with how the `matplotlib.pyplot.plot` signature is)
- `n_out` is a mandatory keyword argument (defining the number of output values)

**Comment by jbednar** [comment]
Thanks. We'll be looking at downsampling options in the coming months, and we'll be sure to check this out!

**Comment by Hoxbro** [comment]
The first two options mentioned in the first comment were implemented in #5552

**Comment by TheoMathurin** [comment]
Alright so the idea is to use `downsample1d` directly instead of `decimate` + some options. It feels a bit like there are two parallel ways to perform downsampling but that's fine with me.

Thanks @Hoxbro!
